invisible_cities/cities/components.py

```diff
@@ -594,12 +594,16 @@ def hits_and_kdst_from_files( paths : List[str]
             except (tb.exceptions.NoSuchNodeError, IndexError):
                 continue
 
-            check_lengths(event_info, hits_df.event.unique())
+            check_lengths(event_info, kdst_df.event.unique())
 
             for evtinfo in event_info:
                 event_number, timestamp = evtinfo.fetch_all_fields()
                 hits = hits_from_df(hits_df.loc[hits_df.event == event_number])
-                yield dict(hits = hits[event_number],
+                if len(hits):
+                    hits = hits[event_number]
+                else:
+                    hits = HitCollection(event_number, timestamp, [])
+                yield dict(hits = hits,
```

Comment:
For `if len(hits)`, the `True` condition occurs under any number of rows within `hits` as long as they aren't zero, which returns `False`.  This feels a bit obtuse, perhaps a comment to explain that the code is checking for hitless events would be useful?